src/Advection/vector_invariant_advection.jl

```diff
@@ -3,17 +3,28 @@ using Oceananigans.Operators
 struct EnergyConservingScheme end
 struct EnstrophyConservingScheme end
 
-struct VectorInvariant{S}
+struct VectorInvariant{S, CA}
     scheme :: S
-end
+    child_advection :: CA
 
-VectorInvariant(; scheme::S = EnstrophyConservingScheme()) where S = VectorInvariant{S}(scheme)
+    function VectorInvariant{S}(scheme::S, child_advection::CA) where {S, CA}
+        return new{S, CA}(scheme, child_advection)
+    end
+end
 
-const VectorInvariantSchemes = Union{VectorInvariant, WENOVectorInvariant}
+VectorInvariant(; scheme::S = EnstrophyConservingScheme()) where S = VectorInvariant{S}(scheme, nothing)
 
 const VectorInvariantEnergyConserving = VectorInvariant{<:EnergyConservingScheme}
 const VectorInvariantEnstrophyConserving = VectorInvariant{<:EnstrophyConservingScheme}
 
+const WENOVectorInvariantVort = Union{WENOVectorInvariantVort3, WENOVectorInvariantVort5}
+const WENOVectorInvariantVel  = Union{WENOVectorInvariantVel3,  WENOVectorInvariantVel5}
+const WENOVectorInvariant{FT, XT, YT, ZT, XS, YS, ZS, VI} = 
+    Union{WENO3{FT, XT, YT, ZT, XS, YS, ZS, VI}, 
+          WENO5{FT, XT, YT, ZT, XS, YS, ZS, VI}} where {FT, XT, YT, ZT, XS, YS, ZS, VI<:SmoothnessStencil}
```

Comment:
Why is `WENOVectorInvariant` identified as a `WENO` with a `SmoothnessStencil`? This doesn't make sense to me. 

The design proposed by #2454 proposed to disentangle the reconstruction scheme from the "formulation":

```julia
struct VectorInvariantMomentumAdvection
     vorticity_stencil
     vorticity_reconstruction
end
```

Therefore "WENOVectorInvariant" is `VectorInvariantMomentumAdvection` with `vorticity_reconstruction::WENO5`.

Reply:
I didn t get to that part yet

Reply:
Okay!